### PR TITLE
Remove sidecar after Istio uninstallation

### PR DESCRIPTION
**Pull request by barchw**
**Description**

Changes proposed in this pull request:

- Trigger restart of injected pods after uninstallation of Istio

**Related issue(s)**

- https://github.com/kyma-project/istio/issues/107

### CODE REVIEW

File: pkg/lib/sidecars/remove/remove_test.go [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+
+var _ = ReportAfterSuite("custom reporter", func(report ginkgotypes.Report) {
+	tests.GenerateGinkgoJunitReport("remove-sidecar-suite", report)
+})
+
+var _ = Describe("Remove Sidecar", func() {
+	ctx := context.TODO()
+	logger := logr.Discard()
+
+	It("should rollout restart Deployment if the pod with sidecar is owned by one", func() {
```
[the remark this answers is not in the document]
Comment (barchw) [reply pkg/lib/sidecars/remove/remove_test.go]: https://github.com/kyma-project/istio/pull/111/commits/4565e81c49707279dd03c279ab9fe641471fb3e6

File: pkg/lib/sidecars/remove/remove.go [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+import (
+	"context"
+
+	"github.com/go-logr/logr"
+	podInfo "github.com/kyma-project/istio/operator/pkg/lib/sidecars/pods"
+	"github.com/kyma-project/istio/operator/pkg/lib/sidecars/restart"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type SidecarRemover struct {
```
[the remark this answers is not in the document]
Comment (barchw) [reply pkg/lib/sidecars/remove/remove.go]: https://github.com/kyma-project/istio/pull/111/commits/4565e81c49707279dd03c279ab9fe641471fb3e6

**Review by triffer — APPROVED** [review]

**Pull request merged**